src/rowbinary/de.rs

```diff
@@ -146,14 +146,79 @@ impl<'cursor, 'data> Deserializer<'data> for &mut RowBinaryDeserializer<'cursor,
         visitor.visit_byte_buf(self.read_vec(size)?)
     }
 
+    #[inline]
+    fn deserialize_identifier<V: Visitor<'data>>(self, visitor: V) -> Result<V::Value> {
+        self.deserialize_u8(visitor)
+    }
+
     #[inline]
     fn deserialize_enum<V: Visitor<'data>>(
         self,
-        name: &'static str,
+        _name: &'static str,
         _variants: &'static [&'static str],
-        _visitor: V,
+        visitor: V,
     ) -> Result<V::Value> {
-        panic!("enums are unsupported: `{name}`");
+        struct Access<'de, 'cursor, 'data> {
+            deserializer: &'de mut RowBinaryDeserializer<'cursor, 'data>,
+        }
+        struct VariantDeserializer<'de, 'cursor, 'data> {
+            deserializer: &'de mut RowBinaryDeserializer<'cursor, 'data>,
+        }
+        impl<'data> VariantAccess<'data> for VariantDeserializer<'_, '_, 'data> {
+            type Error = Error;
+
+            fn unit_variant(self) -> Result<()> {
+                Ok(())
```

Comment:
It seems to be an error on the definition side (`SomeEnum::A` without any payload), innit?

We should return an error here to avoid the generic "not enough data" error. Probably, it's time to introduce `Error::Unsupported` instead of panics, but up to you here.

Reply:
Added `Error::Unsupported` and used it in this case. The rest of the panics can be changed as a follow-up.